prow/cmd/deck/job_history.go

```diff
@@ -211,7 +211,7 @@ func (bucket blobStorageBucket) listSubDirs(ctx context.Context, prefix string)
 
 // Lists all keys with given prefix.
 func (bucket blobStorageBucket) listAll(ctx context.Context, prefix string) ([]string, error) {
-	it, err := bucket.Opener.Iterator(ctx, fmt.Sprintf("%s://%s/%s", bucket.storageProvider, bucket.name, prefix), "")
+	it, err := bucket.Opener.Iterator(ctx, fmt.Sprintf("%s://%s/%s/", bucket.storageProvider, bucket.name, prefix), "")
```

Comment:
nit, I know that currently prefix doesn't have trailing slash, what if by any chance it has, would double slash still work?

/lgtm
/approve
/hold

Reply:
@chaodaiG no doesn't work, I've updated the PR to only add the suffix if it doesn't exist yet